src/java.base/share/classes/java/util/AbstractMap.java

```diff
@@ -304,32 +311,24 @@ public void clear() {
 
     // Views
 
-    /**
-     * Each of these fields are initialized to contain an instance of the
-     * appropriate view the first time this view is requested.  The views are
-     * stateless, so there's no reason to create more than one of each.
-     *
-     * <p>Since there is no synchronization performed while accessing these fields,
-     * it is expected that java.util.Map view classes using these fields have
-     * no non-final fields (or any fields at all except for outer-this). Adhering
-     * to this rule would make the races on these fields benign.
-     *
-     * <p>It is also imperative that implementations read the field only once,
-     * as in:
-     *
-     * <pre> {@code
-     * public Set<K> keySet() {
-     *   Set<K> ks = keySet;  // single racy read
-     *   if (ks == null) {
-     *     ks = new KeySet();
-     *     keySet = ks;
-     *   }
-     *   return ks;
-     * }
-     *}</pre>
-     */
-    transient Set<K>        keySet;
-    transient Collection<V> values;
+    private static final long VIEWS
+            = UNSAFE.objectFieldOffset(AbstractMap.class, "views");
+
+    // Todo: Investigate if we can make this field final
+    // Set via Unsafe
+    private transient Views<K , V> views;
```

Comment:
Note that this field still disqualifies `ImmutableCollections.Map*` from being `@ValueBased`.